Digizuite/Metadata/MetaFieldService.cs

```diff
@@ -41,7 +44,19 @@ public class MetaFieldService : IMetaFieldService
                 _logger.LogError("Failed to load metafields", nameof(response), response);
                 throw new Exception("Failed to load metafields: " + response);
             }
-            
+
+            var languageId = await _damAuthenticationService.GetLanguageId();
+
+            foreach (var field in response.Data!)
+            {
+                var label = field.Labels.SingleOrDefault(l => l.Value.LanguageId == languageId);
```

Comment:
Have you actually tested this against a 5.5 and 5.6? 

Additionally I'm pretty sure we need to return a full instance per label to have the same behavior as 5.5 had for 5.6.

Reply:
Yes, tested with 5.5 and 5.6
Also we have just the same code [here](https://github.com/Digizuite/Digizuite.SDK/blob/master/Digizuite/Extensions/MetadataValueServiceExtensions.cs#L89)

Reply:
I'm pretty sure those are alright because they request by LabelId, and thus it is only for one language. Did you have multiple languages enabled in the DC's you tested against? Since that is where this becomes very relevant. In 5.5 we send a "field" per language per field, where in 5.6 we only send one field per field, irrespective of the number of languages.

Reply:
yes, I have 3 languages active in DC
ok, here is a screenshot of what is happening when I test against 5.6

![image](https://user-images.githubusercontent.com/10358557/164637576-61eba356-6ca5-49cd-8bfc-5b3107b953ed.png)

Reply:
I think Rasmus meant against 5.5 where the Labels array looks different.

But perhaps there is no path between the two versions that is sensible?

Reply:
@madjufra for 5.5.* we do not alter  field.LangaugeId, field.LabelId and field.Label, since labels array is empty
please see screenshot
branch 5.5.0, labels array is empty, so label.Value will be null and so value of field.LabelId  (and others) won't be changed 
(technically it will be overwritten with same value)

![image](https://user-images.githubusercontent.com/10358557/164645291-01a72643-746c-48dc-8c2c-28d3b9c39e5a.png)

Reply:
I'm not talking about the properties themselves, but rather about the number of elements in the response in the first place. If this is to be truly compatible, that number has to be the same before and after the update when this method returns. Right now I have a feeling that it isn't, since the code you added explicitly filters it to only the currently active language.

Reply:
There is no code is legacyservice which allow to have in response both labels array and field.LangaugeId, field.LabelId, field.Label  be initialized with values at the same time

Reply:
@zlepper finally got you point, yes the number of elements will be different, if this might be a problem , we can add missing elements

Reply:
@zlepper finally got you point, so to fix this we have to add more elements to response.

Reply:
Exactly, you basically need to "create" a field per label if any labels are returned. If there is no labels we are talking with a 5.5 and can just return the response as-is.

Reply:
@zlepper updated, please check it out